README.Rmd

```diff
@@ -47,72 +58,82 @@ To install the development version of the package, use:
 remotes::install_github("reconhub/reportfactory")
 ```
 
-Note that this requires the package `remotes` installed.
+
 
 ## Quick start
 
 ### Step 1 - Create a new factory
-Create and open a new factory. Here, we create the factory with a default template but stay in our current activities (set `move_in` to TRUE to switch projects). Check out more templates at [reconhub/report_factories_templates](https://github.com/reconhub/report_factories_templates).
+Create and open a new factory. Here, we create the factory with mostly the
+default settings but stay in our current working directory (set `move_in` to 
+TRUE to switch directories).
+
 
 ```{r creation}
 library(reportfactory)
-factory_proj <- "new_factory"
-new_factory(file.path(factory_proj), include_template = TRUE, move_in = FALSE)
+new_factory("my_factory", path = tempdir())
+```
+
+```{r, setwd, include=FALSE}
+knitr::opts_knit$set(root.dir = fs::path(tempdir(), "my_factory"))
 ```
 
 ### Step 2 - Add your reports
-Here we've already created some with the `include_template` being set to TRUE (the default). The helper functions below show the state of the factory.
+Here we've already created some with most of the default arguments being set to
+TRUE (the default). These default settings include both an example report and
+some associated data (`report_sources/example_report.Rmd` and 
+`data/raw/example_data.csv`).  The helper functions below show the state of the
+factory.
 
 ```{r listing}
-list_reports(factory_proj)
-list_outputs(factory_proj)
-list_deps(factory = factory_proj) # list all needed packages
+list_reports()       # list all available report sources
+list_deps()          # list all of the dependencies of the reports
+list_outputs()       # currently empty
 ```
 
 ### Step 3 - Compile report(s)
 
-#### Option 1 - A single report
-The `compile_report()` function can be used to compile a report using its name, or a partial match for its name. This is useful when you're actively working on developing a single report.
+The `compile_reports()` function can be used to compile a report using regular
+expressions matched against the full filename of reports within the factory. 
 
-```{r compile_one, eval=FALSE}
-compile_report("example_report_2019-01-31.Rmd", factory = factory_proj, quiet = TRUE)
-```
+This ability to use of regular expressions is useful when you're actively
+working on developing your reports but once the factory is setup we recommend
+passing full filenames to the function so it is always clear what will be built.
 
-Use helper functions to see progress.
-```{r compile_one_results}
-list_outputs(factory_proj)
+```{r compile}
+compile_reports( 
+  reports = "example_report.Rmd"
+)
 ```
 
```

Comment:
Suggest adding:
Note that `reports` can also be an `integer` or a `logical` vector, in which case it is interpreted as a subset of files output by `list_reports()`. For instance:
* `reports = c(1, 3)` will compile the first and third reports listed by `list_reports()`
* `reports = TRUE` will compile all reports

Reply:
closed with https://github.com/reconhub/reportfactory/pull/68/commits/8a9da6afe3f55c2610c00e801e0dfb8830e70b4e